File: mobile/ios/OmenIOS/OmenIOS/DesignSystem/OmenBadge.swift
import SwiftUI

enum OmenBadgeTone { case success, neutral, risk, live, stub, mock, unavailable }

/// Registry §3.1 semantic status/data-source label. The label always accompanies its color.
struct OmenBadge: View {
    let label: String
    let tone: OmenBadgeTone

    private var foreground: Color {
        switch tone {
        case .success: return OmenColor.Data.riskLow
        case .neutral: return OmenColor.textSecondary
        case .risk: return OmenColor.Data.riskHigh
        case .live: return OmenColor.Data.dataLive
        case .stub: return OmenColor.Data.dataStub
        case .mock: return OmenColor.Data.dataMock
        case .unavailable: return OmenColor.Data.dataUnavailable
        }
    }

    var body: some View {
        Text(label)
            .omenTextStyle(OmenTypography.chip)
            .foregroundStyle(foreground)
            .padding(.horizontal, OmenSpacing.step8)
            .padding(.vertical, OmenSpacing.step4)
            .background(tone == .neutral ? OmenColor.surface3 : foreground.opacity(0.15))
            .clipShape(Capsule())
    }
}

File: mobile/android/core/designsystem/src/main/kotlin/com/slopssaloon/omen/core/designsystem/component/OmenBadge.kt
package com.slopssaloon.omen.core.designsystem.component

import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.slopssaloon.omen.core.designsystem.theme.OmenTheme

/** Registry §3.1 Badge: semantic status/data-source labels, never a standalone color signal. */
enum class OmenBadgeTone { Success, Neutral, Risk, Live, Stub, Mock, Unavailable }

@Composable
fun OmenBadge(
    label: String,
    tone: OmenBadgeTone,
    modifier: Modifier = Modifier,
) {
    val colors = OmenTheme.color
    val foreground = when (tone) {
        OmenBadgeTone.Success -> colors.data.riskLow
        OmenBadgeTone.Neutral -> colors.textSecondary
        OmenBadgeTone.Risk -> colors.data.riskHigh
        OmenBadgeTone.Live -> colors.data.dataLive
        OmenBadgeTone.Stub -> colors.data.dataStub
        OmenBadgeTone.Mock -> colors.data.dataMock
        OmenBadgeTone.Unavailable -> colors.data.dataUnavailable
    }
    val background = if (tone == OmenBadgeTone.Neutral) colors.surface3 else foreground.copy(alpha = 0.15f)

    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(999.dp),
        color = background,
        contentColor = foreground,
    ) {
        Text(
            text = label,
            modifier = Modifier.padding(horizontal = OmenTheme.spacing.step8, vertical = OmenTheme.spacing.step4),
            style = OmenTheme.typography.chip.toTextStyle(),
        )
    }
}
